test/e2e/common/node/pod_resize.go

```diff
@@ -71,1205 +71,543 @@ func offsetMemory(index int64, value string) string {
+
+				ginkgo.By("deleting pod")
+				podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
+
+			},
+				// All tests will perform the requested resize, and once completed, will roll back the change.
+				// This results in the coverage of both increase and decrease of resources.
+				ginkgo.Entry("cpu increase, then decrease", increasedCPU, originalMem),
+				ginkgo.Entry("mem increase, then decrease", originalCPU, increasedMem),
+				ginkgo.Entry("cpu & mem increase, then decrease", increasedCPU, increasedMem),
+				ginkgo.Entry("increase cpu & decrease mem, then decrease cpu & increase mem", increasedCPU, reducedMem),
+			)
+		},
+			ginkgo.Entry("no restart, no initContainers, no extendedResources", v1.NotRequired, v1.NotRequired, false, false, false),
+			ginkgo.Entry("no restart, no initContainers, with extendedResources", v1.NotRequired, v1.NotRequired, false, false, true),
+			ginkgo.Entry("no restart, has initContainers, resize initContainers", v1.NotRequired, v1.NotRequired, true, true, true),
+			ginkgo.Entry("mem restart policy", v1.NotRequired, v1.RestartContainer, false, false, false),
+			ginkgo.Entry("cpu restart policy", v1.RestartContainer, v1.NotRequired, false, false, false),
+			ginkgo.Entry("both cpu and mem restart policy", v1.RestartContainer, v1.RestartContainer, false, false, false),
+			ginkgo.Entry("both cpu and mem restart policy, has initContainers", v1.RestartContainer, v1.RestartContainer, true, false, false),
+			ginkgo.Entry("both cpu and mem restart policy, has initContainers, resize initContainers", v1.RestartContainer, v1.RestartContainer, true, true, false),
+			ginkgo.Entry("both cpu and mem restart policy, has initContainers, with extendedResources", v1.RestartContainer, v1.RestartContainer, true, false, true),
+		)
+	})
+
+	ginkgo.DescribeTable("guaranteed pods with multiple containers", func(ctx context.Context, expectedContainers []podresize.ResizableContainerInfo) {
+		podClient := e2epod.NewPodClient(f)
+		originalContainers := makeGuaranteedContainers(3, v1.NotRequired, v1.NotRequired, false, false, originalCPU, originalMem)
+		for i := range originalContainers {
+			originalContainers[i].CPUPolicy = nil
+			originalContainers[i].MemPolicy = nil
+		}
+
+		ginkgo.By("creating and verifying pod")
+		newPod := createAndVerifyPod(ctx, f, podClient, originalContainers)
+
+		ginkgo.By("patching and verifying pod for resize")
+		patchAndVerify(ctx, f, podClient, newPod, originalContainers, expectedContainers, "resize")
+
+		// Resize has been actuated, test the reverse operation.
+		rollbackContainers := createRollbackContainers(originalContainers, expectedContainers)
+		ginkgo.By("patching and verifying pod for rollback")
+		patchAndVerify(ctx, f, podClient, newPod, expectedContainers, rollbackContainers, "rollback")
+
+		ginkgo.By("deleting pod")
+		podClient.DeleteSync(ctx, newPod.Name, metav1.DeleteOptions{}, f.Timeouts.PodDelete)
```

Comment:
Is this logic common among the different tables? If so, extract it to a shared function?

Reply:
done